raiden/network/rpc/client.py

```diff
@@ -154,15 +160,29 @@ def dependencies_order_of_build(target_contract, dependencies_map):
     return order
 
 
+def monkey_patch_web3(web3, client):
+    try:
+        # we use a PoA chain for smoketest, use this middleware to fix this
+        web3.middleware_stack.inject(geth_poa_middleware, layer=0)
```

Comment:
Is it okay to inject the poa middleware if we are on ropsten or mainnet? Because this code is called from the ctor of JSONRPCClient.

Reply:
Yes, it's a no-op when not using a geth PoA chain.

Reply:
Thanks. That was my biggest concern on this PR.